#! /bin/sh -e

if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $pdir -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch $pdir -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# DP: XXX patch for builtin echo

--- bash-2.02.1.orig/builtins/echo.def
+++ bash-2.02.1/builtins/echo.def
@@ -163,5 +163,10 @@
   if (display_return)
     putchar ('\n');
   fflush (stdout);
+  if (ferror(stdout))
+    {
+      clearerr(stdout);
+      return (EXECUTION_FAILURE);
+    }
   return (EXECUTION_SUCCESS);
 }


